satpy/readers/hdf4_caliopv3.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+CALIOP v3 HDF4 reader
+"""
+
+import logging
+import os.path
+import re
+
+from datetime import datetime
+from pyhdf.SD import SD, SDC
+
+from satpy.dataset import Dataset
+from satpy.readers.file_handlers import BaseFileHandler
+
+logger = logging.getLogger(__name__)
+
+
+class HDF4BandReader(BaseFileHandler):
+
+    def __init__(self, filename, filename_info, filetype_info):
+        super(HDF4BandReader, self).__init__(filename,
+                                             filename_info,
+                                             filetype_info)
+        self.lons = None
+        self.lats = None
+        self._start_time = None
+        self._end_time = None
+
+        self.filename = filename
+        self.get_filehandle()
+
+        self._start_time = filename_info['start_time']
+
+        logger.debug('Retrieving end time from metadata array')
+        self.get_end_time()
+
+    def get_end_time(self):
+        """ Get observation end time from file metadata
+        """
+        mda_dict = self.filehandle.attributes()
+        core_mda = mda_dict['coremetadata']
+        end_time_str = self.parse_metadata_string(core_mda)
```

Comment:
For parsing the metadata string, you could use hdf_eos.py's EOS metadata parser.